builtins/amp-img.js

```diff
@@ -47,12 +47,21 @@ export class AmpImg extends BaseElement {
 
   /** @override */
   mutatedAttributesCallback(mutations) {
-    if (mutations['src'] !== undefined || mutations['srcset'] !== undefined) {
+    let mutated = false;
+    if (mutations['srcset'] !== undefined) {
+      // `srcset` mutations take precedence over `src` mutations.
       this.srcset_ = srcsetFromElement(this.element);
-      // This element may not have been laid out yet.
-      if (this.img_) {
-        this.updateImageSrc_();
-      }
+      mutated = true;
+    } else if (mutations['src'] !== undefined) {
+      // If only `src` is mutated, then ignore the existing `srcset` attribute
+      // value (may be set automatically as cache optimization).
+      this.srcset_ = srcsetFromSrc(this.element.getAttribute('src'));
```

Comment:
Nit: it's weird that we pass in `element` for `srcsetFrom` but not for `srcsetFromSrc` (I'm ignoring the `Element` suffix). Really is should be `srcsetFromSrcset` and `srcsetFromSrc`, passing in `element` in both cases.

Reply:
`srcsetFromSrcset` sounds even weirder IMO due to overloading of "srcset".